Add v23.1.18 release notes (#18444)
cockroachdb/docs: src/current/_includes/releases/v23.1/v23.1.18.md

## v23.1.18

Release Date: April 9, 2024

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v23-1-18-sql-language-changes">SQL language changes</h3>

- Fixed an oversight where CockroachDB was allowing mutation statements (such as [`UPDATE`](https://www.cockroachlabs.com/docs/v23.1/update) or [`DELETE`](https://www.cockroachlabs.com/docs/v23.1/delete)) and locking statements (such as [`SELECT ... FOR UPDATE`](https://www.cockroachlabs.com/docs/v23.1/select-for-update)) in implicit single-statement [transactions](https://www.cockroachlabs.com/docs/v23.1/transactions) using [`AS OF SYSTEM TIME`](https://www.cockroachlabs.com/docs/v23.1/as-of-system-time). [#120159][#120159]

<h3 id="v23-1-18-operational-changes">Operational changes</h3>

- The `admission.kv.bulk_only.enabled` [cluster setting](https://www.cockroachlabs.com/docs/v23.1/cluster-settings#setting-admission-kv-bulk-only-enabled), when set to `true`, only skips store-level [admission control](https://www.cockroachlabs.com/docs/v23.1/admission-control) for normal priority work, and continues to subject the work to CPU admission control. [#119173][#119173]

> <h3 id="v23-1-18-db-console-changes">[DB Console](https://www.cockroachlabs.com/docs/v23.1/ui-overview) changes</h3>

Author: Fixed - thanks!

> - Resolved an issue where clusters with multiple stores per node could list inaccurate region/node information on the [Databases Page](https://www.cockroachlabs.com/docs/v23.1/ui-databases-page). [#120234][#120234]

Author: Added!

> - Fixed a bug where the timeseries graphs shown on the [SQL Activity Statement Details Page](https://www.cockroachlabs.com/docs/v23.1/ui-statements-page) in [DB Console](https://www.cockroachlabs.com/docs/v23.1/ui-overview) were not rendering properly. This involved fixing a bug related to setting the time range of the charts. [#121783][#121783]

Author: ✅

<h3 id="v23-1-18-bug-fixes">Bug fixes</h3>

- Fix a bug where running [`RESTORE`](https://www.cockroachlabs.com/docs/v23.1/restore) on certain [backups](https://www.cockroachlabs.com/docs/v23.1/backup-and-restore-overview) would open a very large number of connections to the [backup storage provider](https://www.cockroachlabs.com/docs/v23.1/use-cloud-storage). [#119925][#119925]

> - If a user has the `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.1/security-reference/authorization#managing-privileges), then they should not be able to see constants inside of queries that originate from other [users](https://www.cockroachlabs.com/docs/v23.1/security-reference/authorization#roles) in the [`SHOW SESSIONS`](https://www.cockroachlabs.com/docs/v23.1/show-sessions) result. The constants should be redacted. The redaction was not occurring; this is now fixed. [#119963][#119963]

Reviewer: ```suggestion
- A user with the `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.1/security-reference/authorization#managing-privileges) can no longer see constants inside of queries that originate from other [users](https://www.cockroachlabs.com/docs/v23.1/security-reference/authorization#roles) in the [`SHOW SESSIONS`](https://www.cockroachlabs.com/docs/v23.1/show-sessions) result. Previously, this redaction did not occur. [#119963][#119963]
```

Author: ✅

> - The [`SHOW QUERIES`](https://www.cockroachlabs.com/docs/v23.1/show-queries) and [`SHOW STATEMENTS`](https://www.cockroachlabs.com/docs/v23.1/show-statements) statements were incorrectly requiring that the user had the `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.1/security-reference/authorization#managing-privileges). However, a user should always be able to view their own queries, even without this privilege. This is now fixed. [#119963][#119963]

Reviewer: ```suggestion
- Previously, the [`SHOW QUERIES`](https://www.cockroachlabs.com/docs/v23.1/show-statements#aliases) and [`SHOW STATEMENTS`](https://www.cockroachlabs.com/docs/v23.1/show-statements) statements incorrectly required the user to have the `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` [privilege](https://www.cockroachlabs.com/docs/v23.1/security-reference/authorization#managing-privileges). However, a user should always be able to view their own queries, even without these privileges. This is now fixed. [#119963][#119963]
```

link to `SHOW QUERIES` was broken.

Author: ✅

- Fixed a bug in which it was possible to `SET transaction_read_only = false` during an [`AS OF SYSTEM TIME`](https://www.cockroachlabs.com/docs/v23.1/as-of-system-time) transaction. [#120159][#120159]
- Fixed a slow memory leak that could accumulate when opening many new [connections](https://www.cockroachlabs.com/docs/v23.1/connection-pooling). The bug was present in v22.2.9+ and v23.1+. [#120242][#120242]

> - Users will no longer see `VIEW` type tables being shown in the [Databases Page](https://www.cockroachlabs.com/docs/v23.1/ui-databases-page) in [DB Console](https://www.cockroachlabs.com/docs/v23.1/ui-overview). Previously, these tables would be listed with no info, only displaying errors. [#120213][#120213]

Reviewer: ```suggestion
- Users will no longer see [views](https://deploy-preview-18441--cockroachdb-docs.netlify.app/docs/v23.2/views) displayed on the [**Databases** page](https://www.cockroachlabs.com/docs/v23.1/ui-databases-page#databases) in [DB Console](https://www.cockroachlabs.com/docs/v23.1/ui-overview). Previously, views would be listed with no information, only displaying errors. [#120213][#120213]
```

Author: ✅

<h3 id="v23-1-18-operational-changes">Operational changes</h3>

- The `admission.kv.bulk_only.enabled` [cluster setting](https://www.cockroachlabs.com/docs/v23.1/cluster-settings#setting-admission-kv-bulk-only-enabled), when set to `true`, only skips store-level [admission control](https://www.cockroachlabs.com/docs/v23.1/admission-control) for normal priority work, and continues to subject the work to CPU admission control. [#119173][#119173]

<h3 id="v23-1-18-db-console-changes">DB Console changes</h3>

- Resolved an issue where clusters with multiple [stores](https://www.cockroachlabs.com/docs/v23.1/cockroach-start#store) per node could list inaccurate region/node information on the [Databases Page](https://www.cockroachlabs.com/docs/v23.1/ui-databases-page). [#120234][#120234]
- Fixed a bug where the [timeseries graphs shown on the **SQL Activity Statement Fingerprint** page](https://www.cockroachlabs.com/docs/v23.1/ui-statements-page#charts) in [DB Console](https://www.cockroachlabs.com/docs/v23.1/ui-overview) were not rendering properly. This involved fixing a bug related to setting the time range of the charts. [#121783][#121783]

<h3 id="v23-1-18-bug-fixes">Bug fixes</h3>

> - Fixed a bug where, on long-running [sessions](https://www.cockroachlabs.com/docs/v23.1/show-sessions) that issue many (hundreds of thousands or more) [transactions](https://www.cockroachlabs.com/docs/v23.1/transactions), CockroachDB could previously "leak" reported memory usage (as accounted by the internal memory accounting system, the limit for which is configured via the [`--max-sql-memory` flag](https://www.cockroachlabs.com/docs/v23.1/cockroach-start#general)). This, in turn, could result in the error message `"root: memory budget exceeded"` for other queries. The bug was present in version 23.1.17 and is now fixed. [#121874][#121874]

Reviewer: ```suggestion
- Previously, on long-running [sessions](https://www.cockroachlabs.com/docs/v23.1/show-sessions) that issue many (hundreds of thousands or more) [transactions](https://www.cockroachlabs.com/docs/v23.1/transactions), CockroachDB could leak reported memory usage (as accounted by the internal memory accounting system, the limit for which is configured via the [`--max-sql-memory` flag](https://www.cockroachlabs.com/docs/v23.1/cockroach-start#general)). This bug, in turn, could result in the error message `"root: memory budget exceeded"` for other queries. The bug was present in version 23.1.17 and is now fixed. [#121874][#121874]
```
